File: dodam-core/src/main/java/b1nd/dodamcore/schedule/domain/exception/ScheduleExceptionCode.java
package b1nd.dodamcore.schedule.domain.exception;

import b1nd.dodamcore.common.exception.ExceptionCode;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;

@RequiredArgsConstructor
public enum ScheduleExceptionCode implements ExceptionCode {

    NOT_FOUND(HttpStatus.NOT_FOUND, "없는 일정");

    private final HttpStatus status;
    private final String message;

    @Override
    public HttpStatus getHttpStatus() {
        return this.status;
    }

    @Override
    public String getExceptionName() {
        return this.name();
    }

    @Override
    public String getMessage() {
        return this.message;
    }

}

File: services/service-neis/src/main/kotlin/com/b1nd/dodamdodam/neis/domain/schedule/exception/ScheduleExceptionCode.kt
package com.b1nd.dodamdodam.neis.domain.schedule.exception

import com.b1nd.dodamdodam.core.common.exception.ExceptionCode
import org.springframework.http.HttpStatus

enum class ScheduleExceptionCode(
    override val status: HttpStatus,
    override val message: String
) : ExceptionCode {
    SCHEDULE_NOT_FOUND(HttpStatus.NOT_FOUND, "시간표 정보를 찾을 수 없어요."),
    COMCIGAN_API_FAILED(HttpStatus.INTERNAL_SERVER_ERROR, "컴시간 API 호출에 실패했어요."),
    COMCIGAN_SCHOOL_NOT_FOUND(HttpStatus.NOT_FOUND, "컴시간에서 학교를 찾을 수 없어요."),
    ;
}
